src/psyclone/psyGen.py

```diff
@@ -5596,6 +5626,20 @@ def datatype(self):
         '''
         return self._datatype
 
+    @datatype.setter
+    def datatype(self, value):
+        ''' Setter for Symbol datatype.
+
+        :param str value: new value for datatype.
+
+        :raises NotImplementedError: if the specified data type is invalid.
+        '''
+        if value not in Symbol.valid_data_types:
+            raise NotImplementedError(
+                "Symbol can only be initialised with {0} datatypes but found "
+                "'{1}'.".format(str(Symbol.valid_data_types), value))
```

Comment:
Will this output unicode strings for Python2?

Reply:
As agreed, valid_data_types are strings we create internally and I've now added a check that `value` is a `str`.